src/Gateway/JobGateway.php

```diff
@@ -6,68 +6,63 @@
 
 use CodeRhapsodie\DataflowBundle\Entity\Job;
 use CodeRhapsodie\DataflowBundle\Repository\JobRepository;
-use Doctrine\ORM\EntityManagerInterface;
-use Doctrine\ORM\Query;
+use Doctrine\DBAL\Query\QueryBuilder;
+use function Doctrine\DBAL\Query\QueryBuilder;
 
 class JobGateway
 {
-    /** @var EntityManagerInterface */
-    private $em;
     /** @var JobRepository */
     private $jobRepository;
 
-    public function __construct(EntityManagerInterface $em, JobRepository $jobRepository)
+    public function __construct(JobRepository $jobRepository)
     {
-        $this->em = $em;
         $this->jobRepository = $jobRepository;
     }
 
-    public function find(int $id): Job
+    public function find(int $id): ?Job
     {
         return $this->jobRepository->find($id);
     }
 
     public function findForScheduled(int $id): iterable
     {
-        return $this->jobRepository->findBy(['scheduledDataflow' => $id], ['requestedDate' => 'desc'], 20);
+        $qb = $this->jobRepository->createQueryBuilder();
+        $qb->andWhere($qb->expr()->eq('scheduled_dataflow_id', $qb->createNamedParameter($id, \PDO::PARAM_INT)))
+            ->orderBy('requested_date', 'desc')
+            ->setMaxResults(20)
+        ;
+        $stmt = $qb->execute();
+        while (false !== ($row = $stmt->fetch(\PDO::FETCH_ASSOC))) {
+            yield $row;
```

Comment:
The return value was an iterable of `Job` before, we should keep the same return type.

Add the docblock with `@return iterable|Job[]` to make this clearer.

Reply:
This function is used only to read data in the Twig template. Twig can use an array instead of an object because you don't change the job state.
If you want to return an object some changes are needed in DataflowBundle to hydrate objects.

Reply:
The main point is: we shouldn't be passing raw database data around, but business objects instead.

Reply:
1. this function is never used! remove it?
1. What is the goal?
    1. get data for display?
    1. get all business objects to work and call business method?

Reply:
If this function is unused, there is no reason to keep it.

The role of this class is to retrieve Jobs from the underlying storage. How they are used by the caller is unimportant here.

Reply:
Ok to remove function.

Please continue the discussion in the appropriate place.